### DISCUSSION TITLE
How to access the indicators time-series?

### DISCUSSION THREAD

**Discussion by felonfusk** [kernc/backtesting.py#1044] [General]
Following up on my previous post: https://github.com/kernc/backtesting.py/discussions/1024
After some research, however, I am still struggling to find a good way to access the indicators time series.
I was able to find them in the below example but they are arrays and I have no idea the chart legend correctly identify the labels like "SMA(C,10)".
```
from backtesting import Backtest, Strategy
from backtesting.lib import crossover
from backtesting.test import SMA, GOOG
class SmaCross(Strategy):
    def init(self):
        price = self.data.Close
        self.ma1 = self.I(SMA, price, 10)
        self.ma2 = self.I(SMA, price, 20)

    def next(self):
        if crossover(self.ma1, self.ma2):
            self.buy()
        elif crossover(self.ma2, self.ma1):
            self.sell()


bt = Backtest(GOOG, SmaCross, commission=.002,
              exclusive_orders=True)
stats = bt.run()
indicators = bt._results._strategy._indicators  # this will return a list of indicators, but no labels / metadata?
bt.plot()
```

Questions:
1. How do I at least find the labels like those in the chart legend?
2. Is it possible to provide an easy way to generate the one large DataFrame with all the time series combined (data, indicators, trades, positions, etc.)?

**Comment by peacerider** [comment]
Certainly, I don't know which is which from list of "indicators ".

**Comment by kernc** [comment]
1. Try [`.name` indicator attribute](https://github.com/kernc/backtesting.py/blob/1c12381d66ab45539deefb8a3cae51df86c609b3/backtesting/_util.py): `indicator_name = bt._results._strategy._indicators[0].name`.
